kit/dapp/src/lib/utils/zod/amount.ts

```
@@ -0,0 +1,395 @@
+  title: "Whole number amount",
+  description: "A positive whole number amount (no decimals)",
+  examples: [1, 100, 1000],
+  type: "integer",
+  minimum: 1,
+});
+
+z.globalRegistry.add(cryptoAmount(), {
+  id: "crypto_amount",
+  title: "Cryptocurrency amount",
+  description: "A cryptocurrency amount with up to 18 decimal places",
+  examples: [1.5, 0.000000000000000001, 1000.123456789012345678],
```

Biome (1.9.4)

[error] This number literal will lose precision at runtime.

The value at runtime will be 1000.1234567890124

(lint/correctness/noPrecisionLoss)